beacon-chain/sync/decode_pubsub.go

```diff
@@ -22,6 +24,15 @@ func (s *Service) decodePubsubMessage(msg *pubsub.Message) (proto.Message, error
 	if err != nil {
 		return nil, err
 	}
+	// Specially handle subnet messages.
+	switch {
+	case strings.Contains(topic, p2p.GossipAttestationMessage):
+		topic = p2p.GossipTypeMapping[reflect.TypeOf(&ethpb.Attestation{})]
+		// Given that both sync message related subnets have the same message name, we have to
+		// differentiate them below.
+	case strings.Contains(topic, p2p.GossipSyncCommitteeMessage) && !strings.Contains(topic, p2p.SyncContributionAndProofSubnetTopicFormat):
+		topic = p2p.GossipTypeMapping[reflect.TypeOf(&ethpb.SyncCommitteeMessage{})]
+	}
```

Comment:
Add default case and return error please

Reply:
hmm i think this switch is meant to only catch these 3 topics. If the topic is not in the switch cases, we shouldn't error but instead handle normally

Reply:
+1, it is only meant for subnets